mitiq/mitiq_qiskit/tests/test_zne_mitiq_qiskit.py

```diff
@@ -55,10 +82,64 @@ def measure(circuit, qid) -> QuantumCircuit:
     return circuit
 
 
+def run_with_noise(
+    circuit: QuantumCircuit,
+    noise: float,
+    shots: int,
+    seed: Optional[int] = None,
+) -> float:
+    """Runs the quantum circuit with a depolarizing channel noise model.
+    Args:
+        circuit: Ideal quantum circuit.
+        noise: Noise constant going into `depolarizing_error`.
+        shots: The Number of shots to run the circuit on the back-end.
+        seed: Optional seed for qiskit simulator.
+    Returns:
+        expval: expected values.
+    """
+    # initialize a qiskit noise model
+    noise_model = NoiseModel()
+
+    # we assume a depolarizing error for each gate of the standard IBM basis
+    # set (u1, u2, u3)
+    noise_model.add_all_qubit_quantum_error(
+        depolarizing_error(noise, 1), ["u1", "u2", "u3"]
+    )
+
+    # execution of the experiment
+    job = qiskit.execute(
+        circuit,
+        backend=QASM_SIMULATOR,
+        basis_gates=["u1", "u2", "u3"],
+        # we want all gates to be actually applied,
+        # so we skip any circuit optimization
+        optimization_level=0,
+        noise_model=noise_model,
+        shots=shots,
+        seed_simulator=seed,
+    )
+    results = job.result()
+    counts = results.get_counts()
+    expval = counts["0"] / shots
+    return expval
+
+
 def qiskit_executor(qp: QPROGRAM, shots: int = 500) -> float:
     return run_with_noise(qp, noise=BASE_NOISE, shots=shots, seed=1)
 
 
+# TODO: Troubleshoot the problems in this executor.
+#       Causes test failures and time increase to 40+ mins
+# def qiskit_executor(qp: QPROGRAM, shots: int = 500) -> float:
+#   expectation = execute_with_shots_and_noise(
+#         qp,
+#         shots=shots,
+#         obs=ONE_QUBIT_GS_PROJECTOR,
+#         noise_model=initialized_depolarizing_noise(BASE_NOISE)
+#     )
+#   return expectation
```

Comment:
```suggestion
def qiskit_executor(qp: QPROGRAM, shots: int = 500) -> float:
    expectation = execute_with_shots_and_noise(
        qp,
        shots=shots,
        obs=ONE_QUBIT_GS_PROJECTOR,
        noise_model=initialized_depolarizing_noise(BASE_NOISE),
        seed=1,
    )
    return expectation
```
@rmlarose could you take a look at this executor locally and provide some feedback? I'm not totally sure where I'm going wrong. I included `seed_simulator=1` in `execute_with_shots_and_noise` and that passes all tests, should we add it as an arg like in the previous `run_with_noise`?

That said, the test run still took ~40 minutes. Still troubleshooting that now, but I assume I'm missing something that could be obvious.

Reply:
Oh, just saw that @andreamari is the main reviewer here, so tagging in as well. I see that the time increase is coming from the projection with so many shots (qiskit_utils line 167-168):
```
    # we need to modify the circuit to measure obs in its eigenbasis
    # we do this by appending a unitary operation
    # obtains a U s.t. obs = U diag(eigvals) U^dag
    eigvals, U = np.linalg.eigh(obs)
    circ.unitary(np.linalg.inv(U), qubits=range(circ.num_qubits))
```
I'm unsure what the solution should be, but I have a test notebook if anyone wants it.

Thanks!